### PR TITLE
Copy `cortex/pkg/storage/bucket` package dependency into Loki

### PR DESCRIPTION
**Pull request by DylanGuedes**
**What this PR does / why we need it**:
As a part of removing Loki dependency on cortex packages, we decided to copy the `storage/bucket` package from cortex to Loki itself.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

File: pkg/storage/bucket/client.go [hunk 1]
```diff
@@ -7,16 +7,16 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/cortexproject/cortex/pkg/util"
```
Comment (kavirajk) [inline pkg/storage/bucket/client.go:10]: Now we can use `loki/pkg/util` :)

File: pkg/storage/bucket/s3/config.go [hunk 2]
```diff
@@ -7,13 +7,13 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/cortexproject/cortex/pkg/util"
```
Comment (kavirajk) [inline pkg/storage/bucket/s3/config.go:10]: Same here :)

**Review by owen-d — APPROVED** [review]

**Review by kavirajk — APPROVED** [review]
LGTM :tada:

**Comment by DylanGuedes** [comment]
fyi: pipeline is failing for an unrelated reason.

**Pull request merged**